File: 0791-custom-sort-string/0791-custom-sort-string.java
class Solution {
    public String customSortString(String order, String s) {
        Map<Character, Integer> letters = new HashMap<>();
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < order.length(); i++) {
            letters.put(order.charAt(i), 0);
        }

        for (int i = 0; i < s.length(); i++) {
            if (letters.containsKey(s.charAt(i))) {
                letters.put(s.charAt(i), letters.get(s.charAt(i)) + 1);
            }
        }

        for (int i = 0; i < order.length(); i++) {
            for (int j = 0; j < letters.get(order.charAt(i)); j++) {
                result.append(order.charAt(i));
            }
        }

        for (int i = 0; i < s.length(); i++) {
            if (!letters.containsKey(s.charAt(i))) {
                result.append(s.charAt(i));
            }
        }

        return result.toString();
    }
}

File: 0791-custom-sort-string/0791-custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        letters = {letter: 0 for letter in order}
        result = list()

        for letter in s:
            if letter in letters:
                letters[letter] += 1

        for letter in order:
            for i in range(letters[letter]):
                result.append(letter)

        for letter in s:
            if letter not in letters:
                result.append(letter)

        return ''.join(result)